clippy_lints/src/methods/mod.rs

```diff
@@ -2895,14 +2919,26 @@ fn lint_skip_while_next<'tcx>(
 fn lint_filter_map<'tcx>(
     cx: &LateContext<'tcx>,
     expr: &'tcx hir::Expr<'_>,
-    _filter_args: &'tcx [hir::Expr<'_>],
-    _map_args: &'tcx [hir::Expr<'_>],
+    filter_args: &'tcx [hir::Expr<'_>],
+    map_args: &'tcx [hir::Expr<'_>],
 ) {
-    // lint if caller of `.filter().map()` is an Iterator
-    if match_trait_method(cx, expr, &paths::ITERATOR) {
-        let msg = "called `filter(..).map(..)` on an `Iterator`";
-        let hint = "this is more succinctly expressed by calling `.filter_map(..)` instead";
-        span_lint_and_help(cx, FILTER_MAP, expr.span, msg, None, hint);
+    // is it specifically `.filter(Option::is_some).map(Option::unwrap)`?
+    if option_filter_map::in_scope(filter_args, map_args) {
+        span_lint_and_help(
+            cx,
+            OPTION_FILTER_MAP,
+            expr.span,
+            "`filter` for `Some` followed by `unwrap`",
+            None,
+            "consider using `flatten` instead",
+        );
```

Comment:
This should be `span_lint_and_sugg`. You have to modify the span for this: `filter_span.with_hi(expr.span.hi())`. You can pass the `filter_span` to this function from the colossal match from above.